[CI] Retry docker pulling if failure when testing Antrea upgrade

lzhecheng opened the pull request:
Sometimes antrea upgrade check fails when docker pulling antrea-ubuntu image because of "Error response from daemon: missing or empty Content-Length header". It happens occasionally and retry should help.

https://github.com/vmware-tanzu/antrea/runs/2305288919?check_suite_focus=true
https://github.com/vmware-tanzu/antrea/runs/2305321865?check_suite_focus=true

lzhecheng commented:
cc @gran-vmv

antoninbas reviewed (commented):
I have also observed such failures. It feels like there may be something wrong with the Harbor repo. I know that they switched to using a CDN recently, so that may explain these failures. I reported the issue to the team maintaining the Harbor registry.

Do you think we could:
* update this script as well: https://github.com/vmware-tanzu/antrea/blob/de29b10a239c68e57ea87da1d974cbebae59df4a/ci/kind/test-e2e-kind.sh#L108 (it has been failing quite often in CI, for the same reason)
* add a `sleep 1` in case of a pull failure, before the next try: I have observed that this tends to happen when doing multiple pulls in a row, maybe waiting a bit in case of failure will help

lzhecheng commented:
> I have also observed such failures. It feels like there may be something wrong with the Harbor repo. I know that they switched to using a CDN recently, so that may explain these failures. I reported the issue to the team maintaining the Harbor registry.
> 
> Do you think we could:
> 
> * update this script as well: https://github.com/vmware-tanzu/antrea/blob/de29b10a239c68e57ea87da1d974cbebae59df4a/ci/kind/test-e2e-kind.sh#L108
>    (it has been failing quite often in CI, for the same reason)
> * add a `sleep 1` in case of a pull failure, before the next try: I have observed that this tends to happen when doing multiple pulls in a row, maybe waiting a bit in case of failure will help

Sure, I have also updated `docker pull` in `test-e2e-kind.sh` and added `sleep 1` in both places.
Although there're also many `docker pull` in other CI scripts and build scripts, I think this issue should be fixed by harbor. So let's leave the others as they are for now.

antoninbas reviewed (dismissed):
@lzhecheng these are the 2 main places where I have seen it fail. Has it failed on Jenkins too?
Apparently it may be dependent both on the Harbor version and the Docker version (so it's possible the issue happens more frequently with the Docker version including in the Github Action VM workers).

antoninbas commented on ci/kind/test-e2e-kind.sh:
```diff
@@ -105,7 +105,10 @@ fi
 for image in "${COMMON_IMAGES_LIST[@]}"; do
-    docker pull $image
+    for i in `seq 3`; do
+        sleep 1
```
why not sleep after it fails?

lzhecheng replied:
Moved `sleep 1` after `docker pull`.

lzhecheng commented:
> @lzhecheng these are the 2 main places where I have seen it fail. Has it failed on Jenkins too?
> Apparently it may be dependent both on the Harbor version and the Docker version (so it's possible the issue happens more frequently with the Docker version including in the Github Action VM workers).

I checked Jenkins jobs (IPv4/v6, Windows) and there isn't any failure when `docker pull`.

lzhecheng commented:
/skip-all
